src/check_bgp_protocols.py

```diff
@@ -0,0 +1,124 @@
+"""InnoGames Monitoring Plugins - BIRD BGP Protocols Health Check
+
+This is a Nagios script which queries BIRD to check if all BGP protocols
+are up and with established connections.
+
+Copyright (c) 2020 InnoGames GmbH
+"""
+
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in
+# all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.  IN NO EVENT SHALL
+# THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+# THE SOFTWARE.
+
+import os
+import re
+import sys
+
+from argparse import ArgumentParser
+from subprocess import check_output, TimeoutExpired, PIPE, STDOUT
+
+
+def parse_args():
+    parser = ArgumentParser(
+        description='Check if all BIRD BGP protocols are up and established'
+    )
+    parser.add_argument('path', help='Path to the birdc/birdc6 binary')
+
+    return parser.parse_args()
+
+
+def main():
+    args = parse_args()
+    birdc_path = args.path
+
+    code, reason = check_birdc_peers(birdc_path)
+
+    print(format_nagios_message(code, reason))
+    sys.exit(code)
+
+
+def check_birdc_peers(path):
+    try:
+        protocols = check_output([path, 'show', 'protocols'], stderr=STDOUT)
+    except FileNotFoundError:
+        return (
+            ExitCodes.critical,
+            'The path to the birdc executable does not exist'
+        )
+    except PermissionError:
+        return (
+            ExitCodes.critical,
+            'The given birdc binary is not executable'
+        )
+
+    if not protocols:
+        return (ExitCodes.critical, 'Could not retrieve BIRD\'s output')
+
+    # Matching protocol lines from BIRD:
+    # direct1  Direct   master   up     2020-02-27
+    # b_af     BGP      master   up     2020-02-28  Established
+    # b_lb     BGP      master   start  18:43:46    Active        Socket: conn
+    # b_home   BGP      master   down   2020-02-29
+    # We are extracting the protocol name, state, date/time since last
+    # change and the additional information about the state of the session.
+    r = re.compile(
+        r'^(?P<name>\w+)\s+BGP\s+(?:\w+)\s+(?P<state>\w+)\s+'
+        r'(?P<since>[\d\-:]+)\s*(?P<info>[\w]+)?.*$',
```

Comment:
The `\s*` right before the optional info group needs to be ungreedy. I guess with this regex you could be safe now:
```suggestion
        r'(?P<since>[\d\-:]+)\s*?(?:(?P<info>[\w]+)|$).*?$',
```
And maybe we should consider to rather tokenise it as you also actually want to do it.

Reply:
I applied this change to the regex. 

Regarding the tokenization, the data doesn't use tabs, but spaces instead. Could we still do it reliably?